src/space.cpp

```diff
@@ -1073,6 +1073,12 @@ static iarf_e do_space(chunk_t *first, chunk_t *second, int &min_sp)
       {
          return(IARF_FORCE);
       }
+      if (chunk_is_token(first, CT_ASM_COLON))
```

Comment:
I don't understand how the assembly colom is related with the open bracket? I don't have access to code so I can't check it myself.

Reply:
I just solved my main issue with asm here.
When the '[' follows the ':' in asm. I exclude rule sp_before_square.
Because the ': [' sequence is typically in asm (with the space).

Maybe this should be made more generic. (Check we are in asm in a more generic fashion ...)